dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/visualization/ChartImageGenerator.java

```diff
@@ -432,7 +434,7 @@ private TextTitle getSubTitle(final Visualization visualization )
         valueMap = DimensionalObjectUtils.getSortedKeysMap( valueMap );
 
         List<NameableObject> seriez = new ArrayList<>( visualization.chartSeries() );
-        List<NameableObject> categories = new ArrayList<>( visualization.chartCategory() );
+        List<NameableObject> categories = new ArrayList<>( defaultIfNull( visualization.chartCategory(), emptyList() ) );
```

Comment:
In DHIS2, you can also use `ObjectUtils.firstNonNull` for selecting a non-null object